### DISCUSSION TITLE
POS tagging

### DISCUSSION THREAD

**Discussion by dandiep** [explosion/spacy-llm#412] [Q&A]
Hi, one of the things we're using spacy-transformers currently is POS tagging. Having to have a different model for each language is a huge problem in terms of memory consumption and architecture. Is it possible to use spacy-llm to do POS tagging, e.g. with OpenAI? Or could it be extended to do so?

**Comment by svlandeg** [comment]
Yes - you can write a custom task to do this with `spacy-llm`, it would probably look a bit like the [lemmatizer](https://github.com/explosion/spacy-llm/tree/main/spacy_llm/tasks/lemma), you can see its prompt template [here](https://github.com/explosion/spacy-llm/blob/main/spacy_llm/tasks/templates/lemma.v1.jinja).

In general, it should be relatively straightforward to implement a custom task, see the docs [here](https://spacy.io/api/large-language-models#tasks) and [here](https://github.com/explosion/spacy-llm/tree/main/usage_examples#writing-your-own-task): you need a `generate_prompts` function and a `parse_responses` function. 

Let us know if you run into issues implementing this!
